.github/workflows/main.yml

```diff
@@ -2,10 +2,12 @@ name: Unity Actions
 
 on:
   pull_request: {}
-  push: { branches: [main, master, develop] }
```

Comment:
If you have `pull_request` and `push`, then commits to PRs will trigger tests twice.

After merging something to `main` the tests should ideally run again, and in case a commit needs to be made to main directly it should also run the workflow.

Not sure if you covered these cases in some other way somehow?

Reply:
Since this uses a personal license instead of a professional license, it should be fine if there are two instances of the workflow running at the same time, right? The `pull_request` workflow should use our credentials from our upstream repo, while the `push` workflow would use the user's credentials from their fork. Is there a potential problem with having the two workflow runs?

I've personally found it very useful to have the tests run on every push. It's been especially useful to confirm each commit in a feature branch.

Reply:
I see. I don't know but a pull request from both local and a fork should just work as expected with the `pull_request` trigger. But I could be wrong?

> Is there a potential problem with having the two workflow runs?

Yes actually we're hitting organisation-wide limits a lot with regards to runners, and the amount provided has become tighter as well a few months ago. When docker repo is building it's already stalling a lot for each commit, other repos don't currently have this by themselves, but they would block docker repo workflows.

However as the effect is probably quite minor I have no problem with moving forward with your changes if it solves a problem. Still trying to understand why push would be needed on all branches if a push to a PR already triggers on `pull_request` (as `PR updated`) afaik.